packages/ckeditor5-image/src/image/converters.js

```diff
@@ -77,8 +78,8 @@ export function srcsetAttributeConverter() {
 		}
 
 		const writer = conversionApi.writer;
-		const figure = conversionApi.mapper.toViewElement( data.item );
-		const img = getViewImgFromWidget( figure );
+		const element = conversionApi.mapper.toViewElement( data.item );
+		const img = element.name === 'img' ? element : getViewImgFromWidget( element );
```

Comment:
Can't this check be inside that helper function?

Also, we should use `element.is( 'element', 'img' )`.